### PR TITLE
feat(severity): Create a new "high priority" alert

### LINKED ISSUE #59872: Create a high priority condition for alert rules
Create a new alert rule condition for "high priority" issues - we want a condition called "when a new high priority issue occurs" that will fire on any of:
* when a new issue with high severity is received
    * "high severity" is currently defined as a severity score higher than 0.1, but you might want to double check with Tillman and Jenn to see if that changed
* when an issue escalates for any reason
    * archived -> escalating
    * new low severity issue escalates

See more technical details in the tech spec [here](https://www.notion.so/sentry/Severity-based-Notifications-Tech-Spec-df83dc1696424075bd3a0ba2e565d916?pvs=4#5e21777509004f1db00e97dfa1f483e5) - we already have a filter and a condition (a filter is the "IF" and a condition is the "WHEN" in alert rule creation) for 2/3 of these so you can copy the logic, but you'll need to write a new one for when a new low severity issue escalates. 

Also worth noting that the archive until escalation condition uses [this](https://github.com/getsentry/sentry/blob/master/static/app/views/alerts/rules/issue/ruleNode.tsx#L319-L324) to override the text - I believe Nisanthan plans on removing the feature flags, but be aware of this when looking at the existing code.

This should be behind a separate feature flag from the existing one (`projects:first-event-severity-alerting`). The existing one is managed in [getsentry](https://github.com/getsentry/getsentry/blob/master/getsentry/features.py#L1631-L1638) and flags in specific projects . 

I anticipate questions from customers about why exactly it fired (or did not fire) so if possible, we should include the reason somewhere in the notification.

This is blocked by #60413, though if you really want to get started you could implement 2/3 of the condition beforehand.

### PR DESCRIPTION
**Pull request by snigdhas**
Adds a new alert condition, HIghPriorityIssueCondition, which evaluated to true if:
- an issue is high severity (> 0.1 severity score)
- an archived issue escalatings
- a new issue (< 1 day old) is detected as escalating 

image

image 

Up next - further testing in a personal project, checking the email formatting and reason

Fixes https://github.com/getsentry/sentry/issues/59872

### CODE REVIEW

File: src/sentry/api/endpoints/project_rules_configuration.py [hunk 1]
```diff
@@ -30,6 +30,7 @@ def get(self, request: Request, project) -> Response:
             "organizations:integrations-ticket-rules", project.organization
         )
         has_issue_severity_alerts = features.has("projects:first-event-severity-alerting", project)
+        has_high_priority_issue_alert = features.has("projects:high-priority-alerts", project)
```
Comment (scefali) [inline src/sentry/api/endpoints/project_rules_configuration.py:33]: @snigdhas so we want a separate feature flag than `projects:first-event-severity-new-escalation`? cc @isabellaenriquez

Comment (snigdhas) [reply src/sentry/api/endpoints/project_rules_configuration.py:33]: I think the alert should be separate from the escalation logic - if there's something wrong with the alert we don't need to stop detecting escalation or vice versa 

It _is_ one more thing to keep track of though..

File: src/sentry/rules/conditions/high_priority_issue.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    def is_high_severity(self, state: EventState, group: Optional[Group]) -> bool:
+        if not group:
+            return False
+
+        try:
+            severity = float(group.get_event_metadata().get("severity", ""))
+        except (KeyError, TypeError, ValueError):
+            return False
+
+        return severity >= 0.1
```
Comment (wedamija) [inline src/sentry/rules/conditions/high_priority_issue.py:27]: This value should be a constant. Also, should it be higher than 0.1?

File: src/sentry/rules/conditions/high_priority_issue.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    def passes(self, event: GroupEvent, state: EventState) -> bool:
+        has_issue_priority_alerts = features.has("projects:high-priority-alerts", self.project)
+        if not has_issue_priority_alerts:
+            return False
+
+        is_high_severity = self.is_high_severity(state, event.group)
+        is_escalating = state.has_reappeared or state.has_escalated
```
Comment (wedamija) [inline src/sentry/rules/conditions/high_priority_issue.py:35]: Are we intentionally including regressed issues here? I skimmed the spec but didn't see any mention of that specifically.

File: src/sentry/rules/conditions/high_priority_issue.py (hunk 2 above)
Comment (scefali) [reply src/sentry/rules/conditions/high_priority_issue.py:27]: @wedamija 0.1 is right but it should definitely be a constant

File: src/sentry/rules/conditions/high_priority_issue.py (hunk 3 above)
Comment (snigdhas) [reply src/sentry/rules/conditions/high_priority_issue.py:35]: We use `has_reappeared` for "The issue changes state from archived to escalating" in the ReappearedEventCondition.

File: src/sentry/rules/conditions/high_priority_issue.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            .order_by("-datetime")[:limit]
+            .values_list("group", "datetime", "data")
+        )
+
+        return [
+            ConditionActivity(
+                group_id=a[0], type=ConditionActivityType.REAPPEARED, timestamp=a[1], data=a[2]
+            )
+            for a in activities
+        ]
```
Comment (wedamija) [inline src/sentry/rules/conditions/high_priority_issue.py:62]: Not sure this is comprehensive, I wonder if we should be writing an event for new escalating so you can fetch it here?

**Review by wedamija — APPROVED** [review]

File: src/sentry/rules/conditions/high_priority_issue.py (hunk 4 above)
Comment (snigdhas) [reply src/sentry/rules/conditions/high_priority_issue.py:62]: that could work - I think we do write an activity for Escalating too so I can include for that as well 

https://github.com/getsentry/sentry/blob/8a9f0ad73d075a8fe5d94bbe913d9c885054b0bf/src/sentry/issues/escalating.py#L534

Comment (wedamija) [reply src/sentry/rules/conditions/high_priority_issue.py:62]: I wonder if we should also write an activity here? https://github.com/getsentry/sentry/blob/f6fd15d3a195b11d065344ac3bb8fdc5f6153893/src/sentry/tasks/post_process.py#L1377-L1378

Comment (snigdhas) [reply src/sentry/rules/conditions/high_priority_issue.py:62]: yeah that makes sense to me - I can get that PR up

**Pull request merged**